### PR TITLE
feat: OpenID4VP Proof request Base work and UI

### PR DESCRIPTION
**Pull request by MosCD3**
# Summary of Changes

This PR introducing the flow of accepting a proof request from an issuer using OIDC flow

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/package.json [hunk 1]
```diff
@@ -197,14 +199,14 @@
     "react-native-fs": "^2.16.6",
     "react-native-gesture-handler": "^1.10.3",
     "react-native-get-random-values": "^1.7.0",
-    "react-native-gifted-chat": "*",
```
Comment (cvarjao) [inline packages/legacy/core/package.json:200]: I can understand that `*` is super generic, but any particular reason for setting to a very specific peer dependencies? Usually keep compatibility at major level (starting with `^`) is fine. Similar comment  for `react-native-qrcode-svg`, and `react-native-reanimated` changes below.

Comment (MosCD3) [reply packages/legacy/core/package.json:200]: The "*" caused installing latest stable version, and it caused a whole lot of conflicts with other packages that are mot updated, and to do that in this PR is a big mess. I had to fix the version to get things aligned together and get the app to build and tests to succeed.

Comment (MosCD3) [reply packages/legacy/core/package.json:200]: TBH, I believe leaving yarn to pickup latest updated versions for any packages is recipe for bugs. We have to dedicate PRs to solely update specific packages.

Comment (cvarjao) [reply packages/legacy/core/package.json:200]: we can be specific at `dependencies` but more liberal with `peer dependencies`. Also, dependencies should not be updated as a side-effect, it should be explicit and managed by the lock file.

Comment (cvarjao) [reply packages/legacy/core/package.json:200]: keep in mind that `yarn install` MAY produce unexpected dependency change. `yarn install --immutable` is the safest command to use and void unexpected changes to the lock file

Comment (MosCD3) [reply packages/legacy/core/package.json:200]: I never meant to touch any of that :). It NEVER succeeded until I fixed versions for those packages. Let me know if you have another idea and we can hop on a call and discuss

File: packages/legacy/core/App/modules/openid/resolverProof.tsx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        }
+      } else if (query.request && typeof query.request === 'string') {
+        const _res = {
+          data: query.request,
+          certificate: extractCertificateFromJwt(query.request),
+        }
+        return _res
+      }
+    }
+    return { data: null, certificate: null }
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/modules/openid/resolverProof.tsx:143]: In future, can we prefer using early returns so there isn't as many nested ifs? Like this for example:
```
if (!uri) {
  return { data: null, certificate: null }
}

// rest of conditional logic
```

File: packages/legacy/core/App/screens/Connection.tsx [hunk 3]
```diff
@@ -246,6 +247,11 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
       })
       return
     }
+
+    if ((state.notificationRecord as OpenId4VPRequestRecord).type === 'OpenId4VPRequestRecord') {
+      dispatch({ inProgress: false })
+      navigation.replace(Screens.OpenIDProofPresentation, { credential: state.notificationRecord })
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Connection.tsx:253]: In future, can we avoid passing objects as navigation params? I think react-navigation recommends only using primitives as params, so we could pass a credential ID instead.

**Review by bryce-mcmath — APPROVED** [review]
Approved with some suggestions for future PRs

File: packages/legacy/core/App/screens/Connection.tsx (hunk 3 above)
Comment (MosCD3) [reply packages/legacy/core/App/screens/Connection.tsx:253]: we are working in a model that will create a react-hooks for open-id work, this then will be solved

**Pull request merged**